chroot-bin/gnome-session-wrapper

```diff
@@ -33,6 +33,10 @@ if [ "${DESKTOP_SESSION#cinnamon}" != "$DESKTOP_SESSION" ] && \
     SESSION="$CINNAMON_SESSION"
 fi
 
+SESSION_FILE="/usr/share/$SESSION/sessions/$DESKTOP_SESSION.session"
```

Comment:
Yep, for any combination of GNOME/Unity (gnome-session) or cinnamon (cinnamon-session).